addons/mail/models/ir_mail_server.py

```diff
@@ -13,6 +13,87 @@ class IrMailServer(models.Model):
         inverse_name='mail_server_id',
         string='Mail template using this mail server',
         readonly=True)
+    
+    @api.model
+    def send_email(
+        self,
+        message,
+        mail_server_id=None,
+        smtp_server=None,
+        smtp_port=None,
+        smtp_user=None,
+        smtp_password=None,
+        smtp_encryption=None,
+        smtp_ssl_certificate=None,
+        smtp_ssl_private_key=None,
+        smtp_debug=False,
+        smtp_session=None,
+    ):
+        """Inject autogenerated header for autogoing mails"""
+
+        if not self.env.context.get(
+            "mail_autogenerated_header_disable"
+        ) and self._send_email_set_autogenerated(
+            message,
+            mail_server_id=mail_server_id,
+            smtp_server=smtp_server,
+            smtp_port=smtp_port,
+            smtp_user=smtp_user,
+            smtp_password=smtp_password,
+            smtp_encryption=smtp_encryption,
+            smtp_ssl_certificate=smtp_ssl_certificate,
+            smtp_ssl_private_key=smtp_ssl_private_key,
+            smtp_debug=smtp_debug,
+            smtp_session=smtp_session,
+        ):
+            # MS Exchange's broken version as of
+            # http://blogs.technet.com/b/exchange/archive/2006/10/06/
+            # 3395024.aspx
+            message["Precedence"] = "bulk"
+            message["X-Auto-Response-Suppress"] = "OOF"
```

Comment:
That blog post seems to suggest the second option is sufficient?

```
OOF messages are also not sent as responses for incoming messages with the X-Auto-Response-Suppress:OOF header.
```
Might also make sense to make it `"all"` instead of just "OOF"? As the goal seems to be to never get an auto-response.

Note that we already use X-Auto-Response-Suppress in some places. So either way might be good to consider adding auto-submitted to these.

Referencing a more modern and official documentation would be nice, I've found some more modern blog posts from the microsoft exchange team about this subject

Reply:
seems `Precedence` is [discouraged](https://datatracker.ietf.org/doc/html/rfc2076#section-3.9) anyways.

I find a couple of MS docs linking to https://learn.microsoft.com/en-us/openspecs/exchange_server_protocols/ms-oxcmail/ced68690-498a-4567-9d14-5c01f974d8b1, so this might be the most canonical link.

In my original version I didn't do all because I figured you'd like to receive ie an NDR for marking a mail address as defunct.